08-regression2.Rmd

```diff
@@ -453,41 +463,94 @@ get our test error.
+we likely would have little hope of using that variable to predict the house's sale price
+(barring any future remarkable scientific discoveries about the relationship between 
+the housing market and homeowner icecream preferences). In cases like these, 
+the only option is to obtain measurements of more useful variables.
+
+There are, however, a wide variety of cases where the predictor variables do have a 
+meaningful relationship with the response variable, but that relationship does not fit
+the assumptions of the regression method you have chosen. For example, a dataframe `df`
+with two variables---`x` and `y`---with a nonlinear relationship between the two variables
+will not be correctly captured by simple linear regression:
+<center>
+```{r 08-predictor-design, message = FALSE, warning = FALSE, echo = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Example of a dataset with a nonlinear relationship between the predictor and the response"}
+set.seed(1)
+df = tibble(x = sample(10000, 100, replace=TRUE)/10000)
+df <- df %>% mutate(y = x^3 + 0.2*sample(10000, 100, replace=TRUE)/10000 - 0.1)
+curve_plt <- ggplot(df, aes(x = x, y = y)) +
+  geom_point() +
+  xlab("x") +
+  ylab("y") +
+  geom_smooth(method = "lm", se = FALSE)
+curve_plt
+```
+</center>
+Instead of trying to predict the response `y` using a linear regression on `x`,
+we might have some scientific background about our problem to suggest that `y`
+should be a cubic function of `x`. So before performing regression,
+we might *create a new predictor variable* `z` using the `mutate` function:
+```{r 08-predictor-design-3, message = FALSE, warning = FALSE}
+df <- df %>%
+        mutate(z = x^3)
+```
+Then we can perform linear regression for `y` using the predictor variable `z`:
+<center>
+```{r 08-predictor-design-2, message = FALSE, warning = FALSE, echo = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Relationship between the transformed predictor and the response"}
+curve_plt2 <- ggplot(df, aes(x = z, y = y)) +
+  geom_point() +
+  xlab("z") +
+  ylab("y") +
+  geom_smooth(method = "lm", se = FALSE)
+curve_plt2
+```
+</center>
+In this case, you can see that the transformed predictor `z` helps the 
+linear regression model make more accurate predictions. The process of
+transforming predictors (and potentially combining multiple predictors in the process)
+is known as *feature engineering*. In real data analysis
+problems, you will need to rely on
+a deep understanding of the problem---as well as the wrangling tools 
+from previous chapters---to engineer useful new features that improve
+predictive performance. 
+
+> Note that feature engineering
+> is *part of tuning your model*, and as such you must not use your test data
+> to evaluate the quality of the features you produce. You are free to use
+> cross-validation, though!
 
 ## The other side of regression
 
 So far in this textbook we have used regression only in the context of
```

Comment:
One thing I think we are missing here, is this other side of regression is done when you want to understand, claim causality. You cannot always claim causality of course, but when you are on this other side of regression, this is what you are typically wanting to do.

Reply:
done!